collab-folder/src/core/view.rs

```diff
@@ -64,6 +64,46 @@ impl ViewsMap {
     self.remove_cache_view(parent_id);
   }
 
+  /// why we did not use `move_child`?
+  /// because we need to move child view from one parent to another parent
+  /// but the new parent maybe not view, like move out from view to workspace
+  pub fn move_child_out(&self, parent_id: &str, view_id: &str) {
+    self.container.with_transact_mut(|txn| {
+      self.move_child_out_with_txn(txn, parent_id, view_id);
+    })
+  }
+
+  /// why we did not use `move_child`?
+  /// because we need to move child view from one parent to another parent
+  /// but the old parent maybe not view, like move in view from workspace
+  pub fn move_child_in(&self, parent_id: &str, view_id: &str, prev_id: Option<String>) {
+    self.container.with_transact_mut(|txn| {
+      self.move_child_in_with_txn(txn, parent_id, view_id, prev_id);
+    })
+  }
+
+  /// move child view out
+  pub fn move_child_out_with_txn(&self, txn: &mut TransactionMut, parent_id: &str, view_id: &str) {
```

Comment:
Do you mean detach the child from the parent? If yes, using detachParentChild name.